subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/BuildLogicLockingIntegrationTest.groovy

```diff
@@ -59,6 +61,9 @@ class BuildSourceBuilderIntegrationTest extends AbstractIntegrationSpec {
             def listener = new TraceListener(pid: pid, timer: timer)
             def manager = gradle.services.get(BuildOperationListenerManager)
             manager.addListener(listener)
+            gradle.buildFinished {
```

Comment:
❓ Can this be replaced with Flow Actions that are CC compatible?

Related spec: [Plugin receives build result](https://docs.google.com/document/d/1nS1JTwdCMQN_0WaaFrSUxoQOj3zOqmKmXxC6k0OXUR0/edit#heading=h.3gba57k2hdtm)

Reply:
This is some debug log that also checks the start of an arbitrary build operation. I don't think flow actions will work. I could register a build service as an operation listener with an internal API, though I didn't want to fix this init script more since it is used in another test only for debugging purposes.

Reply:
Should we raise it somewhere that there is a use-case that works with `buildFinished`, but is not covered with flow actions?